frontend/src/components/UserPhoto.js

```diff
@@ -1,17 +1,20 @@
-import React, {Component, PropTypes} from 'react';
+import React from 'react';
+import avatarByString from '../lib/avatar_by_string';
 
-export default class UserPhoto extends Component {
-  static propTypes = { url: PropTypes.string };
+export default class UserPhoto extends React.Component {
+  static propTypes = {
+    user: React.PropTypes.object,
+    addBadge: React.PropTypes.bool
+  };
 
   static defaultProps = {
-    url: '/static/images/default_avatar.svg',
     addBadge: false
   };
 
   render() {
-    const { className = '', url, addBadge } = this.props;
-    const avatar = (url || '/static/images/default_avatar.svg');
-
+    const { className = '', user, addBadge } = this.props;
+    const userId = (user.id || 0).toString();
```

Comment:
Is toString necessary? We could keep it an Int and simplify the library by modding id with 4.